src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -844,10 +844,11 @@ def copy(
         >>> idx is new_idx
         False
         """
-        # TODO: SNOW-1458120 implement copy
-        WarningMessage.index_to_pandas_warning("copy")
         return Index(
-            self.to_pandas().copy(deep=deep, name=name), convert_to_lazy=self.is_lazy
+            self._query_compiler.copy(),
+            name=name,
+            copy=deep,
```

Comment:
We should never need to make a deep copy since everything from here down should be immutable.

Reply:
Yep, the pandas index docs are unclear on what the function of the `deep` parameter is, and they say that the method should functionally be the same regardless of `deep` being passed, which is weird. How would you suggest I modify this?

Reply:
We probably can just drop the call here and ignore the parameter (same as pandas 😄 ). Would that work for you?
```suggestion
```

Reply:
@sfc-gh-dpetersohn that works for me, just updated it now

Reply:
I am always forgetting to switch my github account. 😞